packages/vc-export/src/KiltAttestationProofV1.ts

```diff
@@ -0,0 +1,633 @@
+  const salt = Object.entries(credential.claimNonceMap)
+    .map(([hsh, slt]) => [hexToU8a(hsh), stringToU8a(slt)])
+    .sort((a, b) => u8aCmp(a[0], b[0]))
+    .map((i) => base58Encode(i[1]))
+  return {
+    type: ATTESTATION_PROOF_V1_TYPE,
+    block,
+    commitments,
+    salt,
+  }
+}
+
+export const proofSchema: JsonSchema.Schema = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  type: 'object',
+  properties: {
+    '@context': {
+      type: 'array',
+      const: DEFAULT_CREDENTIAL_CONTEXTS,
+    },
+    type: {
+      type: 'string',
+      const: ATTESTATION_PROOF_V1_TYPE,
+    },
+    block: {
+      type: 'string',
+    },
+    commitments: {
+      type: 'array',
+      items: { type: 'string' },
+    },
+    salt: {
+      type: 'array',
+      items: { type: 'string' },
+    },
+  },
+
+  additionalProperties: false,
+  required: ['type', 'block', 'commitments', 'salt'],
+}
+
+const schemaValidator = new JsonSchema.Validator(proofSchema)
+
+/**
+ * Validates a proof object against the KiltAttestationProofV1 data model.
+ * Throws if object violates the [[proofSchema]].
+ *
+ * @param proof Proof object to be validated.
+ */
+export function validateStructure(proof: KiltAttestationProofV1): void {
+  const { errors, valid } = schemaValidator.validate(proof)
+  if (!valid)
+    throw new ProofMalformedError(
+      `Object not matching ${ATTESTATION_PROOF_V1_TYPE} data model`,
+      {
+        cause: errors,
+      }
+    )
+}
+
+export interface VerificationResult {
+  verified: boolean
+  errors: Error[]
+}
+
+/**
+ * Normalizes claims in credentialSubject for the commitment scheme of this proof method.
+ * This involves sorting the normalized representation of claims by their blake2b digests.
+ *
+ * @param expandedContents JSON-LD expansion of credentialSubject, where keys are either '@id' or full URIs.
+ * @returns An array of normalized `statements` sorted by their digests, and the sorted array of `digests`.
+ */
+function normalizeClaims(
+  expandedContents: ExpandedContents<CredentialSubject>
+): { statements: string[]; digests: Uint8Array[] } {
+  const statements = Object.entries(expandedContents).map(([key, value]) =>
+    JSON.stringify({ [key]: value }).normalize('NFC')
+  )
+  const statementsAndDigests = statements.map((statement) => ({
+    digest: blake2AsU8a(stringToU8a(statement), 256),
+    statement,
+  }))
+  const sorted = statementsAndDigests.sort((a, b) => {
+    return u8aCmp(a.digest, b.digest)
+  })
+  return {
+    statements: sorted.map(({ statement }) => statement),
+    digests: sorted.map(({ digest }) => digest),
+  }
+}
+
+function makeCommitments(
+  statementDigests: Uint8Array[],
+  salt: Uint8Array[]
+): Uint8Array[] {
+  return statementDigests.map((digest, index) => {
+    // initialize array with 36 + 2 + 64 bytes
+    const bytes = new Uint8Array(102)
+    // add salt to array
+    bytes.set(salt[index])
+    // add bytes 0x30 & 0x78
+    bytes.set([48, 120], 36)
```

Comment:
Just realising that we could potentially pull these bytes into the salt, making it 38 bytes. We'd set them for existing credentials, for new ones I'd just be additional entropy

Reply:
It's '0x' in UTF-8 ;) comes from the hash prefix. A result of the requirement that we want backwards compatibility